workflow/snakemake_rules/download.smk

```diff
@@ -33,7 +33,7 @@ def _infer_decompression(input):
 rule download_sequences:
     message: "Downloading sequences from {params.address} -> {output.sequences}"
     output:
-        sequences = "data/downloaded_{origin}.fasta.gz"
+        sequences = "data/downloaded_{origin}.fasta"
```

Comment:
I don't understand this... we are downloading an `xz` file but save it without suffix?

Reply:
Yes. `xopen` will then read the contents to infer the compression. This allows us to support `gz` and uncompressed FASTA inputs for sequences (as well as anything else `xopen` can read). 

I don't know how controversial this approach is, but there is precedent (e.g. the Auspice JSONs on `nextstrain-data` are `gz` compressed, but the files end in `.json`).

Reply:
It is a shame if there's no way to flag this. I can 100% see this throwing me for an absolute loop - I wouldn't expect a file to be compressed without an ending, so I'd assume it was getting corrupted somehow....

Reply:
There are alternative approaches (e.g. #565). 

Currently the master branch is working (via https://github.com/nextstrain/ncov/commit/750112e2aa2dabbcb74a2a45933459f80abcda23) so there is no longer urgency for this PR.

Reply:
I definitely like how neat this - it's a nice solution. My only hesitation is that I can completely see myself getting stumped by this in about a month (the absolute limit of my memory for all intents and purposes) - and I'm lucky as I could post on slack and have one of you guys remind me 😅 , but others might also get caught up in this (for the open files). However, I'm happy to defer to what works and what's organized.

Reply:
I can't say I love it... the jsons on `nextstrain-data` have the appropriate content encoding so the browsers handle this correctly. 

But with our 'one workflow does everything' approach, this either requires even more plumbing or automagic solutions like this.

Reply:
I'm also not fully on-board with this approach. The precedent of compressed JSONs without standard compressed file extensions has often caused me problems and I can see this similarly confusing us in the future.

Even more generally, I'm worried about depending on `xopen`-specific functionality when we're planning to move away from `xopen` eventually. If we do go this route, we'll need to make sure we implement content sniffing in our own compression handlers, etc.

Even though it's a pain to update extensions throughout the Snakefile, I like @rneher's fix in 750112e.